cpp/ql/test/library-tests/ir/range-analysis/SimpleRangeAnalysis_tests.cpp

```diff
@@ -759,9 +759,9 @@ unsigned long mult_overflow() {
 unsigned long mult_lower_bound(unsigned int ui, unsigned long ul) {
   if (ui >= 10) {
     range(ui); // $ range=>=10
-    range((unsigned long)ui); // $ range=>=10
-    unsigned long result = (unsigned long)ui * ui; // $ overflow=+
-    range(result); // $ MISSING: range=>=100
```

Comment:
Was the annotation wrong here, or does adding the upperbound here magically fix things?

Reply:
Adding the upper-bound here magically fix things :) Because we can prove that it doesn't overflow we're now able to report a bound.